codegen/module_system.go

```diff
@@ -92,10 +100,68 @@ func NewDefaultModuleSystem(h *PackageHelper) (*module.System, error) {
 		)
 	}
 
-	// TODO: Register endpoint module class and type generators
+	middlewareSpecs := map[string]*MiddlewareSpec{}
```

Comment:
I think the middlewares need to be modeled as a dependency but we can think about that later. It might make sense to treat them as a global grab bag (i.e. a single module of middlewares)